src/Service.Tests/Configuration/HotReload/ConfigurationHotReloadTests.cs

```diff
@@ -301,34 +304,129 @@ public async Task HotReloadConfigRuntimeGQLEnabledEndToEndTest()
+    /// connection string. Lastly, we assert that the first reload fails while the second one succeeds.
+    /// </summary>
+    [TestCategory(MSSQL_ENVIRONMENT)]
+    [TestMethod]
+    public async Task HotReloadConfigConnectionString()
+    {
+        // Arrange
+        _writer = new StringWriter();
+        Console.SetOut(_writer);
+
+        string failedKeyWord = "Unable to hot reload configuration file due to";
+        string succeedKeyWord = "Validated hot-reloaded configuration file";
+
+        // Act
+        // Hot Reload should fail here
+        GenerateConfigFile(
+            databaseType: DatabaseType.MSSQL,
+            connectionString: "");
+        System.Threading.Thread.Sleep(8000);
+
+        // Log that shows that hot-reload was not able to validate properly
+        string failedConfigLog = $"{_writer.ToString()}";
+
+        // Hot Reload should succeed here
+        GenerateConfigFile(
+            databaseType: DatabaseType.MSSQL,
+            connectionString: $"{ConfigurationTests.GetConnectionStringFromEnvironmentConfig(TestCategory.MSSQL).Replace("\\", "\\\\")}");
+        System.Threading.Thread.Sleep(8000);
+
+        HttpResponseMessage restResult = await _testClient.GetAsync("/rest/Book");
+
+        // Log that shows that hot-reload validated properly
+        string succeedConfigLog = $"{_writer.ToString()}";
+
+        // Assert
+        Assert.IsTrue(failedConfigLog.Contains(failedKeyWord));
+        Assert.IsTrue(succeedConfigLog.Contains(succeedKeyWord));
+        Assert.AreEqual(HttpStatusCode.OK, restResult.StatusCode);
+    }
+
+    /// <summary>
+    /// /// (Warning: This test only currently works in the pipeline due to constrains of not
+    /// being able to change from one database type to another, under normal circumstances
+    /// hot reload allows changes from one database type to another)
+    /// Hot reload the configuration file so that it changes from one database type to another.
+    /// Then it hot reloads once more to the original database type. We assert that the
+    /// first reload fails while the second one succeeds.
+    /// </summary>
+    [TestCategory(MSSQL_ENVIRONMENT)]
+    [TestMethod]
+    public async Task HotReloadConfigDatabaseType()
+    {
+        // Arrange
+        _writer = new StringWriter();
+        Console.SetOut(_writer);
+
+        string failedKeyWord = "Unable to hot reload configuration file due to";
+        string succeedKeyWord = "Validated hot-reloaded configuration file";
+
+        // Act
+        // Hot Reload should fail here
+        GenerateConfigFile(
+            databaseType: DatabaseType.PostgreSQL,
+            connectionString: $"{ConfigurationTests.GetConnectionStringFromEnvironmentConfig(TestCategory.POSTGRESQL).Replace("\\", "\\\\")}");
+        System.Threading.Thread.Sleep(5000);
+
+        // Log that shows that hot-reload was not able to validate properly
+        string failedConfigLog = $"{_writer.ToString()}";
```

Comment:
once the failedConfigLog is gathered, can you please clear up the writer's contents so that we dont check for old contents as well?